packages/opentelemetry-instrumentation-anthropic/opentelemetry/instrumentation/anthropic/__init__.py

```diff
@@ -178,47 +254,213 @@ def wrapper(wrapped, instance, args, kwargs):
     return _with_tracer
 
 
+def _build_from_streaming_response(
+        span,
+        response,
+        instance,
+        kwargs
+):
+    complete_response = {"events": [], "model": "", "usage": {}}
+    for item in response:
+        item_to_yield = item
+        if item.type == 'message_start':
+            complete_response["model"] = item.message.model
+            complete_response["usage"] = item.message.usage
+        elif item.type == 'content_block_start':
+            index = item.index
+            if len(complete_response.get("events")) <= index:
+                complete_response["events"].append({"index": index, "text": ""})
+        elif item.type == 'content_block_delta' and item.delta.type == 'text_delta':
+            index = item.index
+            complete_response.get("events")[index]["text"] += item.delta.text
+        elif item.type == 'message_delta':
+            for event in complete_response.get("events", []):
+                event["finish_reason"] = item.delta.stop_reason
+        yield item_to_yield
+
+    # prompt_usage
+    prompt_tokens = 0
+    if kwargs.get("prompt"):
+        prompt_tokens = instance.count_tokens(kwargs.get("prompt"))
```

Comment:
This call can be costly and increase latency. I'd add a flag for the instrumentation controlling whether this is enabled or not - see the OpenAI instrumentation for an example